### PR TITLE
update: new web agg write

### PR DESCRIPTION
**Pull request by nmmarquez**
New aggregate function which aggregates data by different jurisdictions. The function logic is very similar to `calc_aggregate_counts` except each state has a count for each jurisdiction. The main part of the code is here.

```
ucla_df <- read_scrape_data(
        window = window, all_dates = all_dates, wide_data = FALSE) %>%
        mutate(Web.Group = case_when(
            Jurisdiction == "immigration" ~ "ICE",
            Jurisdiction == "federal" ~ "Federal",
            Age == "Juvenile" ~ "Juvenile",
            Jurisdiction == "state" ~ "Prison",
            Jurisdiction == "psychiatric" ~ "Psychiatric",
            Jurisdiction == "county" ~ "County",
            TRUE ~ NA_character_
        ))
```

Leaving this up as a draft for now to see if we agree on this categorizing and how we want to deal with incorporating MP data.

### CODE REVIEW

File: R/alt_aggregate_counts.R [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+            Jurisdiction == "federal" ~ "Federal",
+            Age == "Juvenile" ~ "Juvenile",
+            Jurisdiction == "state" ~ "Prison",
+            Jurisdiction == "psychiatric" ~ "Psychiatric",
+            Jurisdiction == "county" ~ "County",
+            TRUE ~ NA_character_
+        ))
+
+    fac_long_df <- ucla_df %>%
+        filter(State != "Not Available") %>%
```
Comment (erika-tyagi) [inline R/alt_aggregate_counts.R:42]: Thoughts on leaving these in and just keeping the state as `Not Available`? Means we'd keep both federal / ice facilities that we haven't gotten around to adding yet, but also federal aggregations (e.g. RRCs) where they're cross-state so the actual assigned state is `Not Available`.

**Review by erika-tyagi — COMMENTED** [review]
Yay thank you for doing this! And thank you for adding `max_na_rm` as a function too! A few little thoughts: 
1. Do you think it's worth having a `state` param that works the same way as in the other `calc_aggregate_counts`? Obviously extremely easy to get the `state = F` version from the output of this, so not sure if it's worth it. 
2. Is there a reason you called the web grouping `Prison` instead of `State`? (thinking about federal prisons) 
3. Re: MP – I think we have to assign the whole totals to the state web grouping...? Probably whenever MP value > our value (as opposed to just when we don't have a value) to catch Texas, Ohio, etc., I think? 

One thing I'm still iffy about is rates (literally never sure about rates). For `State`-`Web.Group` combinations where we have `Residents.Population `and/or `Population.Feb20` for ALL (or basically all) facilities, that seems super straightforward. And for combinations where we don't have ANY population data, that also seems like a straightforward NA situation. But when we have population values for some (but not all) facilities in a grouping, do we use the total numerator for the measure but the partial denominator? Even if that overestimates rates? I honestly don't know what we're doing right now on the website to address this, so might be a perfect enemy of good situation.

File: R/alt_aggregate_counts.R (hunk 1 above)
Comment (hjohns12) [reply R/alt_aggregate_counts.R:42]: I like this idea but I'm not sure, practically, whether this matters. If state = "Not Available", would the numbers show up anywhere on the website? I don't think so

Comment (erika-tyagi) [reply R/alt_aggregate_counts.R:42]: I think they actually do show up on the website, just with the jurisdiction where the name of the state would be! 
Screen Shot 2021-06-07 at 12 46 31 PM

Comment (nmmarquez) [reply R/alt_aggregate_counts.R:40]: just checking, since these are aggregated they wouldn't show up here right? this is only for "facilities"

Comment (hjohns12) [reply R/alt_aggregate_counts.R:40]: that's my understanding!

File: R/alt_aggregate_counts.R [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    # read in ucla data and do the appropriate grouping
+    fac_long_df <- read_scrape_data(
+        window = window, all_dates = all_dates, wide_data = FALSE) %>%
+        mutate(Web.Group = case_when(
+            Jurisdiction == "immigration" ~ "ICE",
+            Jurisdiction == "federal" ~ "Federal",
+            Age == "Juvenile" ~ "Juvenile",
+            Jurisdiction == "state" ~ "Prison",
+            Jurisdiction == "psychiatric" ~ "Psychiatric",
+            Jurisdiction == "cou nty" ~ "County",
```
Comment (hjohns12) [inline R/alt_aggregate_counts.R:37]: misspelling here on "county"

**Review by hjohns12 — APPROVED** [review]
Aside from one tiny typo, this all looks good to me! Thank you so much for dealing with all this craziness and using good variable names to wit :)

**Review by erika-tyagi — CHANGES_REQUESTED** [review]
This looks wonderful! Made one suggestion, and I still feel like we should keep the `Not Available` state rows, but overall looks great // thank you so much for working through this!!

**Comment by nmmarquez** [comment]
Added back in the "state not available" and changed the calculation of rate to include values for population in the denominator even when we dont have a numerator. This makes our rates more comparable to Marshall project and makes our denominators more sensible.  
f52500f1-75f2-43e6-a9c2-bbd60bb04ef0

File: R/read_mpap_pop_data.R [hunk 3]
```diff
@@ -0,0 +1,39 @@
+#' Get Feb 20, 2020 population data for state prisons from the MP
+#'
+#' Reads the MP dataset and for each prison jurisdiction pulls the staff and
+#' resident population data closest to Feb 20, 2020.
+#'
+#' @return data frame with jurisdiction level pop data from MP
+#'
+#' @examples
+#' \dontrun{
+#' read_staff_popfeb20()
```
Comment (erika-tyagi) [inline R/read_mpap_pop_data.R:10]: I think this should be `read_mpap_pop_data()`.

File: R/read_staff_popfeb20.R [hunk 4]
```diff
@@ -0,0 +1,25 @@
+#' Get Feb 20, 2020 population data for applicable rows in the fac_data
```
Comment (erika-tyagi) [inline R/read_staff_popfeb20.R:1]: Do we use this function anywhere? // I think it's useful, but curious why it ended up here!

**Review by erika-tyagi — APPROVED** [review]
Yay ty ty lgtm! Added a few last comments, but I don't think they need to preclude merging this. I guess the next step is to update `write_latest_data` // let HO know?

File: R/read_staff_popfeb20.R (hunk 4 above)
Comment (nmmarquez) [reply R/read_staff_popfeb20.R:1]: I wanted to add it in if were to start using staff data denoms anywhere. Not using yet.

**Pull request merged**